File: java/controllers/Controller.java
package controllers;

import logic.GameMaster;
import logic.StateRepository;

import java.util.Collections;
import java.util.List;

public class Controller implements GameController, Connectable {
    private GameMaster gameMaster;
    private UserConnection connection;
    private StateRepository stateRepository;

    private void loadDungeon(String file_path) {

    }

    public void pingEndGame(Integer score) {
        connection.sendGameScore(score);
    }

    @Override
    public void startGame(String filerPath) {

    }

    @Override
    public List<Integer> move(int room) {
        return Collections.emptyList();
    }

    @Override
    public void exitGame() {
        stateRepository.saveGameState(gameMaster.getHero(),gameMaster.getRoom(), gameMaster.getMap().hashCode());
    }
}

File: php/controllers/Controller.php
<?php

require_once 'GameController.php';
require_once 'Connectable.php';
require_once 'GameMaster.php';
require_once 'StateRepository.php';

class Controller implements GameController, Connectable {
    private $gameMaster;
    private $connection;
    private $stateRepository;

    private function loadDungeon($file_path) {
        // Implementation here
    }

    public function pingEndGame($score) {
        $this->connection->sendGameScore($score);
    }

    public function startGame($filePath) {
        // Implementation here
    }

    public function move($room) {
        return [];
    }

    public function exitGame() {
        $this->stateRepository->saveGameState(
            $this->gameMaster->getHero(),
            $this->gameMaster->getRoom(),
            spl_object_hash($this->gameMaster->getMap())
        );
    }
}
?>
